src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const startYear = arguments.length < 2 ? 0 : (century - 1) * 100 + 1;
+  const endYear = arguments.length < 2 ? Infinity : startYear + 99;
+
+  const peopleFromGivenCentury = people.filter(person =>
+    person.died > startYear
+    && person.died < endYear
+    && person.sex === 'm');
+
+  const sumOfAges = peopleFromGivenCentury.reduce((a, b) =>
```

Comment:
Its better to replace `a` & `b` arguments with semantic names like a => totalAge/ageAccumulator, b => person/man/human etc.